File: 682.baseball-game.rb
#
# @lc app=leetcode id=682 lang=ruby
#
# [682] Baseball Game
#

# @lc code=start
# @param {String[]} operations
# @return {Integer}
def cal_points(operations)
  result = []
  operations.each do |ops|
    if ops.to_i != 0
      result << ops.to_i
      next
    end

    case ops
    when '+'
      result << result[-1] + result[-2]
    when 'C'
      result.delete_at(-1)
    when 'D'
      result << result[-1] * 2
    end
  end
  result.sum
end
# @lc code=end

File: 682.baseball-game.go
package main

import (
	"strconv"
)

/*
 * @lc app=leetcode id=682 lang=golang
 *
 * [682] Baseball Game
 */

// @lc code=start
func calPoints(operations []string) int {
	result := []int{}

	for _, ops := range operations {
		if ops == string('+') {
			result = append(result, result[len(result)-1]+result[len(result)-2])
		} else if ops == string('C') {
			result = append([]int{}, result[0:len(result)-1]...)
		} else if ops == string('D') {
			result = append(result, result[len(result)-1]*2)
		} else {
			if value, err := strconv.Atoi(ops); err == nil {
				result = append(result, value)
			}
		}
	}
	var answer int
	for _, v := range result {
		answer += v
	}
	return answer
}

// @lc code=end
